platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemProgressDetails.java

```diff
@@ -26,4 +40,270 @@ public DefaultProblemProgressDetails(InternalProblem problem) {
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "file '" + getPath() + "'";
+        }
+    }
+
+    private static class DevelocityLineInFileLocation extends DevelocityFileLocation implements LineInFileLocation {
+        private final org.gradle.api.problems.LineInFileLocation lineInFileLocation;
+
+        public DevelocityLineInFileLocation(org.gradle.api.problems.LineInFileLocation lineInFileLocation) {
+            super(lineInFileLocation);
+            this.lineInFileLocation = lineInFileLocation;
+        }
+
+        @Override
+        public int getLine() {
+            return lineInFileLocation.getLine();
+        }
+
+        @Nullable
+        @Override
+        public Integer getColumn() {
+            return lineInFileLocation.getColumn() <= 0 ? null : lineInFileLocation.getColumn();
+        }
+
+        @Nullable
+        @Override
+        public Integer getLength() {
+            return lineInFileLocation.getLength() <= 0 ? null : lineInFileLocation.getLength();
+        }
+
+        @Override
+        public String getDisplayName() {
+            String location = getPath() + ":" + getLine();
+            if (getColumn() != null) {
+                location += ":" + getColumn();
+            }
+            if (getLength() != null) {
+                location += ":" + getLength();
+            }
+            return "file '" + location + "'";
+        }
+    }
+
+    private static class DevelocityOffsetInFileLocation extends DevelocityFileLocation implements OffsetInFileLocation {
+
+        private final org.gradle.api.problems.OffsetInFileLocation offsetInFileLocation;
+
+        public DevelocityOffsetInFileLocation(org.gradle.api.problems.OffsetInFileLocation offsetInFileLocation) {
+            super(offsetInFileLocation);
+            this.offsetInFileLocation = offsetInFileLocation;
+        }
+
+        @Override
+        public int getOffset() {
+            return offsetInFileLocation.getOffset();
+        }
+
+        @Override
+        public int getLength() {
+            return offsetInFileLocation.getLength();
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "offset in file '" + getPath() + ":" + getOffset() + ":" + getLength() + "'";
+        }
+    }
+
+    private static class DevelocityTaskLocation implements org.gradle.operations.problems.TaskLocation {
+
+        private final TaskLocation taskLocation;
+
+        public DevelocityTaskLocation(TaskLocation taskLocation) {
+            this.taskLocation = taskLocation;
+        }
+
+        @Override
+        public String getBuildPath() {
+            return taskLocation.getBuildPath();
+        }
+
+        @Override
+        public String getTaskPath() {
+            return taskLocation.getTaskPath();
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "task '" + taskLocation.getBuildTreePath() + "'";
+        }
```

Comment:
💭 I wonder if we want to share the location rendering for locations with unknown type with the console and the report.